modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItDynamicParameterTest.java

```diff
@@ -0,0 +1,165 @@
+    private static final Object UNSUPPORTED_SIGN = new Object();
+
+    @Test
+    void testMetadataTypesForDynamicParameters() {
+        int i = 1;
+        for (ColumnType type : ColumnType.values()) {
+            Object param = generateValueByType(i++, type);
+            if (param == UNSUPPORTED_SIGN) {
+                continue;
+            }
+
+            assertQuery("SELECT typeof(?)").withParams(param).returns(toSqlType(type)).check();
+            ;
+            assertQuery("SELECT ?").withParams(param).returns(param).columnMetadata(new MetadataMatcher().type(type)).check();
+        }
+    }
+
+    @Test
+    public void testDynamicParameters() {
+        assertQuery("SELECT COALESCE(?, ?)").withParams("a", 10).returns("a").check();
+        assertQuery("SELECT COALESCE(null, ?)").withParams(13).returns(13).check();
+        assertQuery("SELECT LOWER(?)").withParams("ASD").returns("asd").check();
+        assertQuery("SELECT ?").withParams("asd").returns("asd").check();
+        assertQuery("SELECT ? + ?, LOWER(?) ").withParams(2, 2, "TeSt").returns(4, "test").check();
+        assertQuery("SELECT LOWER(?), ? + ? ").withParams("TeSt", 2, 2).returns("test", 4).check();
+
+        createAndPopulateTable();
+        assertQuery("SELECT name LIKE '%' || ? || '%' FROM person where name is not null").withParams("go").returns(true).returns(false)
+                .returns(false).returns(false).check();
+
+        assertQuery("SELECT id FROM person WHERE name LIKE ? ORDER BY id LIMIT ?").withParams("I%", 1).returns(0).check();
+        assertQuery("SELECT id FROM person WHERE name LIKE ? ORDER BY id LIMIT ? OFFSET ?").withParams("I%", 1, 1).returns(2).check();
+    }
+
+    // After fix the mute reason need to merge the test with above testDynamicParameters
+    @Disabled("https://issues.apache.org/jira/browse/IGNITE-18258")
+    @Test
+    public void testDynamicParameters2() {
+        assertQuery("SELECT POWER(?, ?)").withParams(2, 3).returns(8).check();
+        assertQuery("SELECT SQRT(?)").withParams(4d).returns(2d).check();
+        assertQuery("SELECT ? % ?").withParams(11, 10).returns(BigDecimal.valueOf(1)).check();
+
+        assertQuery("SELECT id from person where salary<? and id>?").withParams(15, 1).returns(2).check();
+    }
+
+    // After fix the mute reason need to merge the test with above testDynamicParameters
+    @Disabled("https://issues.apache.org/jira/browse/IGNITE-18345")
+    @Test
+    public void testDynamicParameters3() {
+        assertQuery("SELECT LAST_DAY(?)").withParams(Date.valueOf("2022-01-01")).returns(Date.valueOf("2022-01-31")).check();
+        assertQuery("SELECT LAST_DAY(?)").withParams(LocalDate.parse("2022-01-01")).returns(Date.valueOf("2022-01-31")).check();
+    }
+
+    /** Need to test the same query with different type of parameters to cover case with check right plans cache work. **/
+    @Test
+    public void testWithDifferentParametersTypes() {
```

Comment:
why do you use approach with **generateValueByType** in testMetadataTypesForDynamicParameters and not use it here ?

Reply:
it's not require for the case